tests/rptest/tests/strict_data_init_test.py

```diff
@@ -0,0 +1,34 @@
+import re
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import RedpandaService, RESTART_LOG_ALLOW_LIST
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+
+STRICT_DATA_LOG_ALLOW_LIST = RESTART_LOG_ALLOW_LIST + [
+    re.compile(
+        "required by the `storage_strict_data_init` configuration property"),
+    re.compile(".redpanda_data_dir does not exist")
+]
+
+
+class StrictDataInitTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, extra_rp_conf={}, **kwargs)
+        self.rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=1, log_allow_list=STRICT_DATA_LOG_ALLOW_LIST)
+    def test_strict_data_init_enabled(self):
+        target_node = self.redpanda.nodes[0]
+
+        # Enable the `storage_strict_data_init` property
+        # the node should fail to restart with it enabled.
+        self.rpk.cluster_config_set("storage_strict_data_init", "true")
+        self.redpanda.stop_node(target_node)
+        self.redpanda.start_node(target_node, expect_fail=True)
```

Comment:
nit: might be nice to test that it's failing for the reason you expect

Reply:
Good idea, will monitor the log during the restart to ensure the exception I'm expecting is thrown.